modules/wsdata_block/src/Plugin/Block/WSDataBlock.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\wsdata_block\Plugin\Block\WSDataBlock.
+ */
+
+namespace Drupal\wsdata_block\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a 'Green House Gas Emissions' block.
+ *
+ * @Block(
+ *   id = "wsdata_block",
+ *   admin_label = @Translation("Wsdata Block"),
+ *   category = @Translation("wsdata")
+ * )
+ */
+class WSDataBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+
+    // Load the wscall entities.
+    $wscalls = entity_load_multiple('wscall');
```

Comment:
We should write a method for the wscall's to generate their own forms instead of having multiple implementations of essentially the same form.